core/shared/platform/nuttx/nuttx_platform.c

```diff
@@ -55,6 +83,17 @@ os_mmap(void *hint, size_t size, int prot, int flags)
 
     if ((uint64)size >= UINT32_MAX)
         return NULL;
+
+#if (WASM_MEM_DUAL_BUS_MIRROR != 0)
+    if ((prot & MMAP_PROT_EXEC) != 0) {
+        d_addr = malloc((uint32)size);
+        if (d_addr == NULL) {
+            return NULL;
+        }
+        i_addr = d_addr + MEM_DUAL_BUS_OFFSET;
+        return in_ibus_ext(i_addr) ? i_addr : d_addr;
```

Comment:
By a way, not only external RAM(PSRAM) use mirror bus address, but also internal RAM(SRAM) is the same. So maybe you use this solution for internal RAM, too.

Reply:
you mean we should indicate from where to allocate before we do it? and does they have different allocator?

Reply:
Yes, to make nuttx platform easier to use, we could make it support to load AOT to sram and run.